src/ngx_http_set_hashed_upstream.c

```diff
@@ -129,10 +178,35 @@ ngx_http_set_hashed_upstream(ngx_conf_t *cf, ngx_command_t *cmd, void *conf)
         return NGX_CONF_ERROR;
     }
 
+    hashConf = ngx_pcalloc(cf->pool,sizeof(ngx_http_set_hashed_upstream_conf));
+    hashConf->ul = ul;
+    hashConf->hashType = hashType;
+    if(ul->nelts <= 0) {
+        hashConf->nodeLen = 0;
+        hashConf->hashNodes = NULL;
+    } else {
+        hashConf->nodeLen = ul->nelts;
+        hashConf->hashNodes = ngx_pcalloc(cf->pool, sizeof(ngx_http_set_hashed_upstream_consistent_hash_node) *
+                                                    hashConf->nodeLen);
+        for (i = 0; i < hashConf->nodeLen; i++) {
+            hashConf->hashNodes[i].name = ul->elts[i];
+            hashConf->hashNodes[i].hash = ngx_hash_key_lc(hashConf->hashNodes[i].name->data,
+                                                          hashConf->hashNodes[i].name->len);
```

Comment:
One real node should have a number of virtual nodes, usually the number is 160, like nginx chash does: https://github.com/nginx/nginx/blob/master/src/http/modules/ngx_http_upstream_hash_module.c#L299

Otherwise, the consistent hash will have uneven distribution.